autogluon/utils/tabular/ml/models/lr/lr_preprocessing_utils.py

```diff
@@ -0,0 +1,67 @@
+from scipy.sparse import hstack
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.preprocessing import OneHotEncoder
+
+
+class OheFeaturesGenerator(BaseEstimator, TransformerMixin):
+
+    def __init__(self, cats_cols):
+        self._feature_names = []
+        self.cats = cats_cols
+        self.ohe_encs = None
+        self.labels = None
+
+    def fit(self, X, y=None):
+        self.ohe_encs = {f: OneHotEncoder(handle_unknown='ignore') for f in self.cats}
```

Comment:
Why not use the OneHotMergeRaresHandleUnknownEncoder:
https://github.com/awslabs/autogluon/blob/3f2e3b3e46cc10e01017257b46e8f5992cbc6c00/autogluon/utils/tabular/ml/models/tabular_nn/categorical_encoders.py#L321

You can see how it's used for the neural net here:
https://github.com/awslabs/autogluon/blob/79d2f7d60a2eebf0f9aaf1d43186b583f81e703a/autogluon/utils/tabular/ml/models/tabular_nn/tabular_nn_model.py#L689
(it can be be a drop-in replacement for the sklearn OneHotEncoder as it offers the same functionality).  It has the big advantage that it can prevent feature explosion with high-cardinality categorical features (it uses an additional `max_categories` hyperparameter and merges all the rarest categories beyond this value).

Reply:
I tried it, but it was causing memory issues on few datasets.

Reply:
Here you mean memory issues during the preprocessing transform, not during the model training, correct? Or is the issue due to lack of sparse matrix support in the OneHotMergeRaresHandleUnknownEncoder?

For now, it's fine with me to use the standard OHE for this PR.  I can later substitute the OHE with the OneHotMergeRaresHandleUnknownEncoder in a separate PR.

Is there a way for you to help me reproduce this issue for debugging (via email)?